### PR TITLE
Pd control temp

### PR DESCRIPTION
**Pull request by bsaund**
I have several small commits in here. This is my temp branch, so there will be things that wont work.

I added an advancedPdControl.html page to play around with adding new things.
I added some css stuff to get columns working the way I wanted.

Most of my changes are in javascript. I did an auto-indent in here so everything got marked as changed :P. The git code review isn't bad though. Still it'll probably be easiest for you to pull from my branch and just play around with it in your IDE.

### CODE REVIEW

File: tutorials/pdControl/pdControlStyles.css [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#underDampedDescription { visibility: hidden; }
+#overDampedDescription { visibility: hidden; }
+#criticallyDampedDescription { visibility: hidden; }
+
+#dampingLevelDescription {
+}
+
+#pdColumn1 {
+    /*background-repeat: no-repeat;*/
+    width: calc(50% - 450px - 30px);
```
Comment (bsaund) [inline tutorials/pdControl/pdControlStyles.css:12]: The css magic to make the columns work

File: tutorials/pdControl/pdControl.js [hunk 2]
```diff
@@ -25,298 +25,298 @@ var panelPtMass = {
...
-			.attr("cy", this.y);
-			d3.select("#dampingLabel")
-			.text(formatLabelString(this.d))
-			.attr("x", this.x);
-		},
-};
+/*
+ Define a slider class
+ */
+function Slider(domain, range, sliderName, circleName, labelName, initialValue) {
```
Comment (bsaund) [inline tutorials/pdControl/pdControl.js:132]: I am not great at javascript, but this seems to be one way to define a class.

**Comment by bsaund** [comment]
My comments aren't great, I'm just testing out the git code review environment

File: tutorials/pdControl/pdControl.js (hunk 2 above)
Comment (MatthewPeterKelly) [inline tutorials/pdControl/pdControl.js:132]: Nice! I'm also very new at JavaScript, so this is good to know.

File: tutorials/pdControl/pdControlStyles.css (hunk 1 above)
Comment (MatthewPeterKelly) [inline tutorials/pdControl/pdControlStyles.css:12]: That calc() function is neat. Another good trick.

**Comment by MatthewPeterKelly** [comment]
I created a new branch: dev, for testing out code. I manually merged your pdControlTemp branch onto dev.

Nice job cleaning up the JavaScript code. I really like the text that changes the description for damping ratio. The columns are neat, but I'm note sure if they add that much. 

I like the check box for sinusoid motion. We could also add a few other behaviors: circular, random step, figure eight, ...   Maybe a drop-down box?

**Comment by bsaund** [comment]
I just got the automated motion working last night. I definitely want to add more motion types. Radio buttons or dropdown are both fine. I like the check box that hides everything so it is not too cluttered. Long term I'd want the next to flow in/out smoothly when the box is check/unchecked. Also, I want sliders to control speed and possibly other parameters of motion.

The changing text description is obviously a placeholder. My current idea is display the actual equations of motions with the updating numbers based on the sliders.

Another plan is to add sliders for gains and mass and make it so you can toggle between them and freq/damping ratio.

I don't understand you statement "the columns are neat, but I'm not sure they add that much". Which columns? I need the css columns in order to display the text nicely on either side of the panel.

Big picture plan: I like your first Pd page almost as is. I'd vote to remove most of the text though (I just don't think people are going to read text when there is a cool demo to play with). I like the simplicity of the first demo; it is very clear what you should do. We add a link at the bottom of the page to the advanced page.

**Comment by MatthewPeterKelly** [comment]
Awesome! I really like the interface for the automatic motion. It does a great job of demonstrating steady-state error among other things.

I like the idea of including equations. Probably can be done nicely either MathJax of CodeCogs. 

As for the control sliders, I think that it would be a good idea to keep the mass constant so that the bounds on the gains make sense (sense the mass scales the whole problem). On the other hand, I think that it would be awesome to have a toggle between dampingRatio/frequency and proportional/derivative versions of the gains.  One neat way to do it would be to show all four, but at any given time one pair is for input, and the other is for output. A toggle switch could be used to determine which pair is in input/output.

Sorry for the confusion about the columns. What I was trying to say was: "would it be better to try to keep one column, for coding simplicity and so that the tutorial works on a smaller screen or half of a monitor". Now that I see a bit more of the big picture, I think that the columns are good, and it will just look funny if viewed on a really small screen or small window.

The big picture plan sounds good. Once we get a few more things working I'll pull the code for both pages over to the main website. Sorry that I haven't been contributing much! I've been crazy trying to make progress on my research, and keep looking at this stuff with only a little bit of time late at night. I've learned a lot from git collaboration, and reading you code! Hopefully I'll be able to make more contribution to the project soon.

**Comment by bsaund** [comment]
I think it is cool to see how mass scales scales the problem. I also think limiting the acceleration and jerk would be interesting and make the problem more appropriate to real world applications. All this may be too tangential though. There is definitely a tradeoff as more features means more confusion.

I have the same thought for the damping/freq and prop/derivative toggles.

No worries about contributing, research is good. I've had a couple of great nights learning css/js/html which I've wanted to do for a long time.

On my team at work before merging into the main branch we tend to squash similar commits. This makes the revision history a bit easier to read. I think it would be worth it for this, since I made a lot of small commits.

Another big picture item: For the pendulum example I think the code should either be under, or as a separate link. Instead that space could be a graph similar to the Pd example.

**Pull request closed without merging**